plugins/dfa-interpreter.ts

```diff
@@ -1,207 +1,102 @@
-        serialGraph = serialGraph.graph;
-        transferProperties(serialGraph, this);
-
-        this.nodes = serialGraph.nodes.map((oldNode: any) => {
-            const result: Node = {
-                label: oldNode.drawableProperties.Label,
-                parents: [],
-                children: []
-            };
-            transferProperties(oldNode, result);
-            return result;
-        });
+export type Nodes = DFANode
+export type Edges = DFAEdge
+export type Graph = DFAGraph
 
-        // This seems like duplicate code but I'm not sure how to clean it up and combine it with the code above.
-        this.edges = serialGraph.edges.map((oldEdge: any) => {
-            const source = this.nodes[oldEdge.source];
-            const destination = this.nodes[oldEdge.destination];
+export class State {
+    constructor(public active: DFANode,
+        public inputLeft: string,
+        public message: string) {
 
-            const result: Edge = {
-                label: oldEdge.pluginProperties.Symbol,
-                source: source,
-                destination: destination
-            };
-
-            transferProperties(oldEdge, result);
-
-            source.children.push(result);
-            destination.parents.push(result);
-            return result;
-        });
     }
 }
 
-/**
- * This function compiles a DFA.
- */
-export function interpret(graph: Graph): Program {
-    let alphabet = new Set<string>();
-    let acceptStates = [];
-    var startState: Node | null = null;
-
-    for (let edge of graph.edges) {
-        let sym: string = edge.label;
-        // if (!sym)
-        //     throw "Sym is null.";
-        if (sym.length != 1) {
-            throw "Symbols must be one character";
+function isEmpty(label?: string) {
+    if (label != undefined) {
+        if (label != "") {
+            return false;
         }
-        alphabet.add(sym);
     }
 
-    for (let node of graph.nodes) {
-        if (node['Start State']) {
-            if (startState != null) {
-                throw "Too many start states.";
+    return true;
+}
+
+export function start(input: DFAGraph, data: string): State | boolean {
+    let start: DFANode | null = null;
+    const accepts = new Set<DFANode>();
+
+    for (const node of input.nodes){
+        if (node.isStartState) {
+            if (!start) {
+                start = node;
+            } else {
+                throw new Error("Only one start state allowed");
             }
-            startState = node;
         }
-        if (node['Accept State']) {
-            acceptStates.push(node);
+
+        if (node.isAcceptState) {
+            accepts.add(node);
         }
-        let symbols = node.children.map((edge: Edge) => edge.label);
-        let uniqueSymbols = new Set(symbols);
-        if (symbols.length !== uniqueSymbols.size) {
-            throw "This interpreter does not handle NFAs. Non-unique edge label detected.";
+
+        if (node.children) {
```

Comment:
I don't think that this is related to #107. With the version of the serializer that I wrote, children was explicitly set and was never null. I'm not sure if it was changed so that this could be null or if it is an unnecessary check.